### ISSUE TITLE
InvalidOperationException while closing

### ISSUE THREAD

**Issue by Nuklon** [ControlzEx/ControlzEx#74]
I can't reproduce this on my machine, but it gets logged. Doesn't matter much as it's during closing but it should be possible to check first if window is already closed.

I'm using latest ControlzEx pre version.

```
System.InvalidOperationException: Cannot set Visibility or call Show, ShowDialog, or WindowInteropHelper.EnsureHandle after a Window has closed.
at System.Windows.Window.VerifyCanShow() at offset 23
at System.Windows.Window.CoerceVisibility(System.Windows.DependencyObject d, System.Object value) at offset 23
at System.Windows.DependencyObject.ProcessCoerceValue(System.Windows.DependencyProperty dp, System.Windows.PropertyMetadata metadata, System.Windows.EntryIndex& entryIndex, System.Int32& targetIndex, System.Windows.EffectiveValueEntry& newEntry, System.Windows.EffectiveValueEntry& oldEntry, System.Object& oldValue, System.Object baseValue, System.Object controlValue, System.Windows.CoerceValueCallback coerceValueCallback, System.Boolean coerceWithDeferredReference, System.Boolean coerceWithCurrentValue, System.Boolean skipBaseValueChecks) at offset 99
at System.Windows.DependencyObject.UpdateEffectiveValue(System.Windows.EntryIndex entryIndex, System.Windows.DependencyProperty dp, System.Windows.PropertyMetadata metadata, System.Windows.EffectiveValueEntry oldEntry, System.Windows.EffectiveValueEntry& newEntry, System.Boolean coerceWithDeferredReference, System.Boolean coerceWithCurrentValue, System.Windows.OperationType operationType) at offset 649
at System.Windows.DependencyObject.SetValueCommon(System.Windows.DependencyProperty dp, System.Object value, System.Windows.PropertyMetadata metadata, System.Boolean coerceWithDeferredReference, System.Boolean coerceWithCurrentValue, System.Windows.OperationType operationType, System.Boolean isInternal) at offset 682
at System.Windows.DependencyObject.SetValue(System.Windows.DependencyProperty dp, System.Object value) at offset 27
at ControlzEx.Controls.GlowWindow.SetVisibilityIfPossible(System.Windows.Visibility newVisibility) at offset 28
at ControlzEx.Controls.GlowWindow.Update() at offset 132
at ControlzEx.Controls.GlowWindow.<.ctor>b__15_23(System.Object sender, System.EventArgs e) at offset 6
at System.EventHandler.Invoke(System.Object sender, System.EventArgs e)
at System.Windows.Window.OnActivated(System.EventArgs e) at offset 31
at System.Windows.Window.WmActivate(System.IntPtr wParam) at offset 41
at System.Windows.Window.WindowFilterMessage(System.IntPtr hwnd, System.Int32 msg, System.IntPtr wParam, System.IntPtr lParam, System.Boolean& handled) at offset 198
at System.Windows.Interop.HwndSource.PublicHooksFilterMessage(System.IntPtr hwnd, System.Int32 msg, System.IntPtr wParam, System.IntPtr lParam, System.Boolean& handled) at offset 46
at MS.Win32.HwndWrapper.WndProc(System.IntPtr hwnd, System.Int32 msg, System.IntPtr wParam, System.IntPtr lParam, System.Boolean& handled) at offset 48
at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object o) at offset 54
at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate callback, System.Object args, System.Int32 numArgs) at offset 119
at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object source, System.Delegate callback, System.Object args, System.Int32 numArgs, System.Delegate catchHandler) at offset 2
```

**Comment by Nuklon** [comment]
` <PackageReference Include="ControlzEx" Version="4.0.0-alpha0285" />`

**Comment by Nuklon** [comment]
@punker76 Clarified, still needed to get to machine :)

**Comment by batzen** [comment]
There are like twenty checks to prevent this and yet this still happens. I guess we should just catch all exceptions in SetVisibilityIfPossible...

**Comment by punker76** [comment]
@Nuklon Can you say what machine this is? OS? Remote?

**Comment by Nuklon** [comment]
> @Nuklon Can you say what machine this is? OS? Remote?

All Windows 10 (Home/Pro/Enterprise), not used through remote.

**Comment by punker76** [comment]
@Nuklon this has now been fixed (hopefully), and it will be released in the next version of ControlzEx v4.0

**Comment by snowiceyeah** [comment]
The same error has recently occurred, has it been fixed?This error has never appeared before.

**Comment by batzen** [comment]
Which version of ControlzEx are you using?

**Comment by snowiceyeah** [comment]
> Which version of ControlzEx are you using?

3.0.2.4
![image](https://user-images.githubusercontent.com/43429385/58161926-c786c800-7cb3-11e9-882c-9b55f42de724.png)

**Comment by batzen** [comment]
The issue should be fixed in the latest preview of version 4.0.

**Comment by snowiceyeah** [comment]
And I used MahApps.Metro-1.6
Not easy to upgrade

**Comment by snowiceyeah** [comment]
> The issue should be fixed in the latest preview of version 4.0.

Could you please tell me how to fix this bug on my version?
